iq/iq.go

```diff
@@ -0,0 +1,221 @@
+// Copyright 2020 Sonatype Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// Package iq has definitions and functions for processing golang purls with Nexus IQ Server
+package iq
+
+import (
+	"bytes"
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"os"
+	"time"
+
+	"github.com/package-url/packageurl-go"
+	"github.com/sonatype-nexus-community/nancy/buildversion"
+	"github.com/sonatype-nexus-community/nancy/configuration"
+	"github.com/sonatype-nexus-community/nancy/cyclonedx"
+)
+
+const internalApplicationIDURL = "/api/v2/applications?publicId="
+
+const thirdPartyAPILeft = "/api/v2/scan/applications/"
+
+const thirdPartyAPIRight = "/sources/nancy?stageId="
+
+const (
+	pollInterval = 1 * time.Second
+)
+
+type applicationResponse struct {
+	Applications []application `json:"applications"`
+}
+
+type application struct {
+	ID string `json:"id"`
+}
+
+type thirdPartyAPIResult struct {
+	StatusURL string `json:"statusUrl"`
+}
+
+// StatusURLResult is a struct to let the consumer know what the response from Nexus IQ Server was
+type StatusURLResult struct {
+	PolicyAction  string `json:"policyAction"`
+	ReportHTMLURL string `json:"reportHtmlUrl"`
+	IsError       bool   `json:"isError"`
+	ErrorMessage  string `json:"errorMessage"`
+}
+
+var localConfig configuration.Configuration
+
+var statusURLResp StatusURLResult
+
+func getPurls(purls []string) (result []packageurl.PackageURL) {
+	for _, v := range purls {
+		purl, _ := packageurl.FromString(v)
+		result = append(result, purl)
+	}
+
+	return
+}
+
+// AuditPackages accepts a slice of purls, public application ID, and configuration, and will submit these to
+// Nexus IQ Server for audit, and return a struct of StatusURLResult
+func AuditPackages(purls []string, applicationID string, config configuration.Configuration) StatusURLResult {
+	localConfig = config
+
+	if localConfig.User == "admin" && localConfig.Token == "admin123" {
```

Comment:
Maybe pull "admin" and "admin123" out into a const to be used both here and in the Config class??